src/python/grpcio_observability/grpc_observability/_cyobservability.pyx

```diff
@@ -0,0 +1,331 @@
+  CY_CLIENT_ROUNDTRIP_LATENCY = kRpcClientRoundtripLatencyMeasureName
+  CY_CLIENT_SERVER_LATENCY = kRpcClientServerLatencyMeasureName
+  CY_CLIENT_STARTED_RPCS = kRpcClientStartedRpcsMeasureName
+  CY_CLIENT_RETRIES_PER_CALL = kRpcClientRetriesPerCallMeasureName
+  CY_CLIENT_TRANSPARENT_RETRIES_PER_CALL = kRpcClientTransparentRetriesPerCallMeasureName
+  CY_CLIENT_RETRY_DELAY_PER_CALL = kRpcClientRetryDelayPerCallMeasureName
+  CY_CLIENT_TRANSPORT_LATENCY = kRpcClientTransportLatencyMeasureName
+  CY_SERVER_SENT_MESSAGES_PER_RPC = kRpcServerSentMessagesPerRpcMeasureName
+  CY_SERVER_SENT_BYTES_PER_RPC = kRpcServerSentBytesPerRpcMeasureName
+  CY_SERVER_RECEIVED_MESSAGES_PER_RPC = kRpcServerReceivedMessagesPerRpcMeasureName
+  CY_SERVER_RECEIVED_BYTES_PER_RPC = kRpcServerReceivedBytesPerRpcMeasureName
+  CY_SERVER_SERVER_LATENCY = kRpcServerServerLatencyMeasureName
+  CY_SERVER_STARTED_RPCS = kRpcServerStartedRpcsMeasureName
+
+@enum.unique
+class MetricsName(enum.Enum):
+  CLIENT_STARTED_RPCS = _CyMetricsName.CY_CLIENT_STARTED_RPCS
+  CLIENT_API_LATENCY = _CyMetricsName.CY_CLIENT_API_LATENCY
+  CLIENT_SNET_MESSSAGES_PER_RPC = _CyMetricsName.CY_CLIENT_SNET_MESSSAGES_PER_RPC
+  CLIENT_SEND_BYTES_PER_RPC = _CyMetricsName.CY_CLIENT_SEND_BYTES_PER_RPC
+  CLIENT_RECEIVED_MESSAGES_PER_RPC = _CyMetricsName.CY_CLIENT_RECEIVED_MESSAGES_PER_RPC
+  CLIENT_RECEIVED_BYTES_PER_RPC = _CyMetricsName.CY_CLIENT_RECEIVED_BYTES_PER_RPC
+  CLIENT_ROUNDTRIP_LATENCY = _CyMetricsName.CY_CLIENT_ROUNDTRIP_LATENCY
+  CLIENT_SERVER_LATENCY = _CyMetricsName.CY_CLIENT_SERVER_LATENCY
+  CLIENT_RETRIES_PER_CALL = _CyMetricsName.CY_CLIENT_RETRIES_PER_CALL
+  CLIENT_TRANSPARENT_RETRIES_PER_CALL = _CyMetricsName.CY_CLIENT_TRANSPARENT_RETRIES_PER_CALL
+  CLIENT_RETRY_DELAY_PER_CALL = _CyMetricsName.CY_CLIENT_RETRY_DELAY_PER_CALL
+  CLIENT_TRANSPORT_LATENCY = _CyMetricsName.CY_CLIENT_TRANSPORT_LATENCY
+  SERVER_SENT_MESSAGES_PER_RPC = _CyMetricsName.CY_SERVER_SENT_MESSAGES_PER_RPC
+  SERVER_SENT_BYTES_PER_RPC = _CyMetricsName.CY_SERVER_SENT_BYTES_PER_RPC
+  SERVER_RECEIVED_MESSAGES_PER_RPC = _CyMetricsName.CY_SERVER_RECEIVED_MESSAGES_PER_RPC
+  SERVER_RECEIVED_BYTES_PER_RPC = _CyMetricsName.CY_SERVER_RECEIVED_BYTES_PER_RPC
+  SERVER_SERVER_LATENCY = _CyMetricsName.CY_SERVER_SERVER_LATENCY
+  SERVER_STARTED_RPCS = _CyMetricsName.CY_SERVER_STARTED_RPCS
+
+# Delay map creation due to circular dependencies
+_CY_METRICS_NAME_TO_PY_METRICS_NAME_MAPPING = {x.value: x for x in MetricsName}
+
+def cyobservability_init(object exporter) -> None:
+  exporter: grpc_observability.Exporter
+
+  NativeObservabilityInit()
+  _start_exporting_thread(exporter)
+
+
+def _start_exporting_thread(object exporter) -> None:
+  exporter: grpc_observability.Exporter
+
+  global GLOBAL_EXPORT_THREAD
+  global GLOBAL_SHUTDOWN_EXPORT_THREAD
+  GLOBAL_SHUTDOWN_EXPORT_THREAD = False
+  GLOBAL_EXPORT_THREAD = Thread(target=_export_census_data, args=(exporter,))
```

Comment:
Should this be a daemon thread? If not, the thread hanging could prevent the process from exiting?